eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/Abstraction/AzureResponseWIthTypeApi.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.TypeSpec.Generator.Expressions;
+using Microsoft.TypeSpec.Generator.Primitives;
+using Microsoft.TypeSpec.Generator.Snippets;
+using System;
+
+namespace Azure.Generator.Providers.Abstraction
+{
+    /// <summary>
+    /// Represents an Azure response API that can be used to operate with the response from an Azure service.
+    /// </summary>
+    public abstract record AzureResponseWIthTypeApi : ScopedApi, IAzureResponseWithTypeApi
```

Comment:
Can we use a visitor to update the callsite instead of introducing an abstraction?

Reply:
There are visitors methods that allow you to visit individual method statements.